lib/chef/knife/bootstrap.rb

```diff
@@ -638,8 +638,10 @@ def connect!
           EOM
           # FIXME: this should save the key to known_hosts but doesn't appear to be
           config[:ssh_verify_host_key] = :accept_new
-          do_connect(connection_opts(reset: true))
-        elsif ssh? && e.cause && e.cause.class == Net::SSH::AuthenticationFailed
+          conn_opts = connection_opts(reset: true)
+          opts.merge! conn_opts
+          connect!(opts)
+        elsif ssh? && e.class == Train::ClientError && e.message == "Your SSH Agent has no keys added, and you have not specified a password or a key file"
```

Comment:
Error check with string comparison could be risky. Also @ L#660, would you please confirm if this is really the last option.

Reply:
@Nimesh-Msys sure this might be an improvement area for https://github.com/inspec/train